tutorials/nlp/Multitask_Prompt_and_PTuning.ipynb

```diff
@@ -0,0 +1,1184 @@
+   "execution_count": null,
+   "id": "f2c943ba",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from nemo.utils.exp_manager import exp_manager\n",
+    "\n",
+    "# Set name of the experiment \n",
+    "config.name = 'sentiment_intent_slot_p_tuning'\n",
+    "\n",
+    "# Init the experiment manager and view the exp_dir\n",
+    "exp_dir = exp_manager(trainer, config.get(\"exp_manager\", None))\n",
+    "exp_dir = str(exp_dir)\n",
+    "print(exp_dir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5860bd90",
+   "metadata": {},
+   "source": [
+    "We can also set learning hyperparameters as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4c4ec542",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set some of the learning parameters\n",
+    "config.model.optim.lr = 1e-4\n",
+    "config.model.batch_size = 16"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "298b3dce",
+   "metadata": {},
+   "source": [
+    "# Training (P-Tuning)\n",
+    "The only thing left to do is load up the model and begin p-tuning!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b4bda19b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from nemo.collections.nlp.models.language_modeling.megatron_gpt_prompt_learning_model import MegatronGPTPromptLearningModel\n",
+    "\n",
+    "model = MegatronGPTPromptLearningModel(cfg=config.model, trainer=trainer)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2d99f433",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "trainer.fit(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5d5a5798",
+   "metadata": {},
+   "source": [
+    "When training completes, prompt tuned virtual tokens from the prompt encoder are automatically moved to a `prompt_table` where all prompt tuned and p-tuned soft prompts are stored. The LSTM `prompt_encoder` is then removed from the model. This allows us to preserve previously p-tuned soft prompts while still maintaining the ability to add new p-tuned or prompt-tuned soft prompts in the future. The `prompt_table` uses the `taskname` as a key to look up the correct virtual tokens for a specified task."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6aab09d4",
+   "metadata": {},
+   "source": [
+    "# Inference After P-Tuning\n",
+    "One way to run inference after p-tuning or prompt-tuning your model is to call `model.generate()`. `model.generate()` takes in \n",
+    "\n",
+    "- `inputs` which can be either a list of dictionary objects or `.jsonl` files containing dictionary objects, \n",
+    "- `length_params`\n",
+    "- `sampling_params`\n",
+    "\n",
+    "as arguments. More information about the [text generation API can be found here](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/modules/common/transformer/text_generation.py).\n",
```

Comment:
we should mention for most predictive downstream tasks (not text generation tasks), always use `greedy` sampling.